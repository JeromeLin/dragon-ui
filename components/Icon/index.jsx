import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class Icon extends Component {
  render() {
    const {
      type, theme, className, style
    } = this.props;
    const cls = classnames({
      'ui-icon': true,
      [`ui-icon-${type}`]: !!type,
      [`theme-${theme}`]: !!theme,
      [className]: !!className
    });

    return <i className={cls} style={style} />;
  }
}

Icon.propTypes = {
  type: PropTypes.string,
  theme: PropTypes.oneOf(['default', 'info', 'success', 'warning', 'error']),
  style: PropTypes.objectOf(PropTypes.any)
};

Icon.defaultProps = {
  type: '',
  theme: 'default',
  style: {}
};

export default Icon;
