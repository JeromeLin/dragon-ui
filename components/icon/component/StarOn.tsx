import React from 'react';

const SvgStarOn = props => (
  <svg
    width="1em"
    height="1em"
    viewBox="0 0 36 34"
    fill="currentColor"
    stroke="currentColor"
    {...props}
  >
    <path
      d="M25.482 21.062l7.129-7.305-10.06-1.72L17.804 3l-4.744 9.037L3 13.757l7.128 7.305-1.473 10.1 9.15-4.522 9.15 4.522z"


      strokeWidth={2.6}
      fillRule="evenodd"
    />
  </svg>
);

export default SvgStarOn;
