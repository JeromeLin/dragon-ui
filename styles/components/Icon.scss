
@font-face {
  font-family: "dragonicon";
  src: url('#{$base-icon-url}.eot');
  src: url('#{$base-icon-url}.eot?#iefix') format('embedded-opentype'),
    url('#{$base-icon-url}.woff') format('woff'),
    url('#{$base-icon-url}.ttf') format('truetype'),
    url('#{$base-icon-url}.svg#dragonicon') format('svg');
}

.ui-icon {
  font-family: "dragonicon" !important;
  font-style: normal;
  -webkit-font-smoothing: antialiased;
  -webkit-text-stroke-width: 0.2px;
  -moz-osx-font-smoothing: grayscale;
}

.ui-icon-check:before { content: "\e602"; }
.ui-icon-check-round:before { content: "\e60a"; }
.ui-icon-check-round-fill:before { content: "\e609"; }
.ui-icon-close:before { content: "\e605"; }
.ui-icon-close-round:before { content: "\e60e"; }
.ui-icon-close-round-fill:before { content: "\e60b"; }
.ui-icon-info-round:before { content: "\e625"; }
.ui-icon-info-round-fill:before { content: "\e624"; }
.ui-icon-question-round-fill:before { content: "\e618"; }
.ui-icon-question-round:before { content: "\e619"; }
.ui-icon-arrow-top:before { content: "\e600"; }
.ui-icon-arrow-bottom:before { content: "\e601"; }
.ui-icon-arrow-left:before { content: "\e603"; }
.ui-icon-arrow-right:before { content: "\e604"; }
.ui-icon-add:before { content: "\e606"; }
.ui-icon-minus:before { content: "\e607"; }
.ui-icon-date:before { content: "\e63d"; }
.ui-icon-loading:before { content: "\e608"; }

