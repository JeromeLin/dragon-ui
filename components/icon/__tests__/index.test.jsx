import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from '../index';
import StarOn from '../component/StarOn';

describe('Icon', () => {
  it('renders Icons of different theme correctly', () => {
    const wrapper = render(
      <div>
        <Icon type="star-on">default</Icon>
        <Icon theme="primary" type="star-on">info</Icon>
        <Icon theme="success" type="star-on">success</Icon>
        <Icon theme="warning" type="star-on">warning</Icon>
        <Icon theme="danger" type="star-on">danger</Icon>
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Icons of different size correctly', () => {
    const wrapper = render(
      <div>
        <Icon size="xl" theme="primary" type="star-on">xl尺寸</Icon>
        <Icon size="lg" theme="primary" type="star-on">lg尺寸</Icon>
        <Icon theme="primary" type="star-on">默认尺寸</Icon>
        <Icon size="sm" theme="primary" type="star-on">sm尺寸</Icon>
        <Icon size="xs" theme="primary" type="star-on">xs尺寸</Icon>
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('triggers onClick callback correctly on nomarl Icon', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Icon type='star-on' onClick={onClick}>star-on</Icon>,
    );
    wrapper.find('icon').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
