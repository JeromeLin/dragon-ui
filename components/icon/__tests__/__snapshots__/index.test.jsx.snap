// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders Icons of different size correctly 1`] = `
Array [
  <i
    class="zw-icon zw-icon--broadcast zw-icon--primary zw-icon--lg"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--broadcast zw-icon--primary zw-icon--md"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--broadcast zw-icon--primary zw-icon--sm"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>,
]
`;

exports[`Icon renders Icons of different theme correctly 1`] = `
Array [
  <i
    class="zw-icon zw-icon--right zw-icon--primary zw-icon--md"
    theme="primary"
    type="right"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--add zw-icon--success zw-icon--md"
    theme="success"
    type="add"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#add"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--info-round zw-icon--warning zw-icon--md"
    theme="warning"
    type="info-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#info-round"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--wrong-round-fill zw-icon--danger zw-icon--md"
    theme="danger"
    type="wrong-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong-round-fill"
      />
    </svg>
  </i>,
]
`;

exports[`Icon renders Icons using iconfont 1`] = `
Array [
  <i
    class="zw-icon zw-icon--icon-empty_fill zw-icon--primary zw-icon--lg"
    theme="primary"
    type="icon-empty_fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#icon-empty_fill"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--icon-caidan07 zw-icon--primary zw-icon--lg"
    theme="primary"
    type="icon-caidan07"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#icon-caidan07"
      />
    </svg>
  </i>,
]
`;

exports[`Icon renders Icons using svg component 1`] = `
<i
  class="zw-icon zw-icon--primary zw-icon--lg"
  theme="primary"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    viewBox="0 0 32 32"
    width="1em"
  >
    <path
      d="M1 12.376l8.8 9.114L30.431 1.568"
      fill="none"
      stroke="currentColor"
      stroke-width="2.6"
    />
  </svg>
</i>
`;

exports[`Icon renders all Icons 1`] = `
Array [
  <i
    class="zw-icon zw-icon--add zw-icon--primary zw-icon--md"
    theme="primary"
    type="add"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#add"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--arrow-left zw-icon--primary zw-icon--md"
    theme="primary"
    type="arrow-left"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#arrow-left"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--arrow-bottom zw-icon--primary zw-icon--md"
    theme="primary"
    type="arrow-bottom"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#arrow-bottom"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--arrow-right zw-icon--primary zw-icon--md"
    theme="primary"
    type="arrow-right"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#arrow-right"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--broadcast zw-icon--primary zw-icon--md"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--date zw-icon--primary zw-icon--md"
    theme="primary"
    type="date"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#date"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--deletekey zw-icon--primary zw-icon--md"
    theme="primary"
    type="deletekey"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#deletekey"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--info-round zw-icon--primary zw-icon--md"
    theme="primary"
    type="info-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#info-round"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--info-round-fill zw-icon--primary zw-icon--md"
    theme="primary"
    type="info-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#info-round-fill"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--keyboard zw-icon--primary zw-icon--md"
    theme="primary"
    type="keyboard"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#keyboard"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--minus zw-icon--primary zw-icon--md"
    theme="primary"
    type="minus"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#minus"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--question-round zw-icon--primary zw-icon--md"
    theme="primary"
    type="question-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#question-round"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--required zw-icon--primary zw-icon--md"
    theme="primary"
    type="required"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#required"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--right-round-fill zw-icon--primary zw-icon--md"
    theme="primary"
    type="right-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right-round-fill"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--right-round zw-icon--primary zw-icon--md"
    theme="primary"
    type="right-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right-round"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--right zw-icon--primary zw-icon--md"
    theme="primary"
    type="right"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--search zw-icon--primary zw-icon--md"
    theme="primary"
    type="search"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#search"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--time-circle zw-icon--primary zw-icon--md"
    theme="primary"
    type="time-circle"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#time-circle"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--warning-round-fill zw-icon--primary zw-icon--md"
    theme="primary"
    type="warning-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#warning-round-fill"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--wrong-round-fill zw-icon--primary zw-icon--md"
    theme="primary"
    type="wrong-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong-round-fill"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--wrong-round zw-icon--primary zw-icon--md"
    theme="primary"
    type="wrong-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong-round"
      />
    </svg>
  </i>,
  <i
    class="zw-icon zw-icon--wrong zw-icon--primary zw-icon--md"
    theme="primary"
    type="wrong"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong"
      />
    </svg>
  </i>,
]
`;
